### DISCUSSION TITLE
Loading field data from secondary file

### DISCUSSION THREAD

**Discussion by dkaddar** [Nek5000/nekRS#593] [Q&A] [answered]
Hey, I am looking for a way to load the content of a secondary restart file during initialization. In my case, i want to load previously averaged results to compute some fluctuations during a new run.

I guess the iofld class should provide this functionality, but I wasn't quite able to get it to work. When loading the file, I am getting a warning that only part of the data is read (WARNING: restart file has a NSPCAL > LDIMT, read only part of the fld-data!) . Additionally, I was not sure which function one can use to actually access the field data in the udf file.

If someone could help in creating a minimal working example, I would gladly appreciate it.

Below is a code snippet of my current approach:
```
deviceMemory<dfloat> o_field;
...

void UDF_Setup()
{
  o_field.resize(nrs->mesh->Nlocal);
  iofld = iofldFactory::create("nek");
  iofld->open(nrs->mesh, iofld::mode::read, "tavg", 1);

  // I would like to access scalar01 and assign its values to o_field:
  // iofld->addVariable("scalar01", std::vector<deviceMemory<dfloat>>{o_field});
  ...
}
```
Best regards,
Driss

**Comment by stgeke** [comment, marked as answer]
Refer to the `turbPipe` example, where we demonstrate how to write a user-defined field to a file. 
Note, scalar is a 0-based index. If you want to read the first field you have to use `scalar00`.

**Comment by dkaddar** [comment]
Thanks! I simply overlooked calling the process() function.
